testing/sdk_diag.py

```diff
@@ -101,6 +105,32 @@ def handle_test_setup(item: pytest.Item):
     _testlogs_test_index += 1
 
 
+def _task_whitelist_callback(item: pytest.Item):
+    """Returns a callback configured by pytest marker diag_task_whitelist
+    to check if a task is whitelisted, which should be used like this:
+
+    def whitelist_fn(task):
+        return re.match('hello.*', task.name)
+
+    @pytest.mark.diag_task_whitelist.with_args(whitelist_fn)
+    def your_test_here(): ...
+
+    Note that the diag_task_whitelist marker can be used on function, class, or module
+    to be able to hierarchically configure the whitelist.
+    """
+    def _callback(task):
+        if item.get_closest_marker(name='diag_task_whitelist') is None:
+            return True
```

Comment:
Does this mean that we whitelist tasks by default?

Reply:
I changed that, we shouldn't whitelist tasks by default